src/main/java/org/thoughtcrime/securesms/util/ViewUtil.java

```diff
@@ -267,4 +268,14 @@ public static int getStatusBarHeight(@NonNull View view) {
       return result;
     }
   }
+
+  // Checks if a selection is valid for a given Spinner view.
+  // Returns given selection if valid.
+  // Otherwise, to avoid ArrayIndexOutOfBoundsException, 0 is returned, assuming to refer to a good default.
+  public static int checkBounds(int selection, AbsSpinner view) {
+    if (selection < 0 || selection >= view.getCount()) {
+      return 0;
```

Comment:
Should print a stack trace to the log here since something went wrong, maybe even show a toast

Reply:
not idea how to print a stack trace without having an exception object.

but we could log a warning, that is probably good enough

Reply:
i added a commit, the shown warning looks like the following now:

```
index -9 out of bounds of androidx.appcompat.widget.AppCompatSpinner{a8b85c4 VFED..CL. ......I. 0,0-0,0 #7f0a009e app:id/cert_check}
```

Reply:
> not idea how to print a stack trace without having an exception object.

you can simply do
```
new RuntimeException("Selection " + selection + " is not available for this spinner").printStackTrace();
```

> but we could log a warning, that is probably good enough

yes, also good enough

Reply:
btw, printing stacktrace instead of using the logging API is discourage IIRC